### PR TITLE
AMD and CommonJS support and some additional functionality

### PR DESCRIPTION
**Pull request by jhorowitz-firedrum**

### CODE REVIEW

**Comment by jhorowitz-firedrum** [comment]
Bump :)

**Comment by SirCumz** [comment]
MERGE!!!

**Comment by mervick** [comment]
Thanks

**Comment by SirCumz** [comment]
The way emojione lib is loaded should be changed. i cannot use emojione in the first 300 milliseconds because it is not available yet in the window.emojione object because it is loaded trough $.script. maybe the user should be responsible to loaded it in themselves before using emojione-area. just like jquery.

**Comment by jhorowitz-firedrum** [comment]
Tag v3.1.7? ^^;

**Comment by mervick** [comment]
@jhorowitz-firedrum sorry, I don't have much time, 
in fact, I did not even check your code, just fast review and merged.
Of course, i will can create tag. 
Maybe if you interested, I can add you as colobrator for this package.

File: src/wrapper.js [hunk 1]
```diff
@@ -5,10 +5,27 @@
  * Released under the MIT license
  * Date: @DATE
  */
-(function(document, window, $) {
-    'use strict';
+window = ( typeof global !== "undefined" ? global : typeof self !== "undefined" ? self : typeof window !== "undefined" ? window : {} );
+document = window.document || {};
```
Comment (stof) [inline src/wrapper.js]: defining global variables `document` and `window` could have weird side-effects on other scripts. You should avoid this IMO

**Pull request merged**